### PR TITLE
Use Java 8 stream API

### PR DESCRIPTION
**Pull request by zhenxiao**

### CODE REVIEW

**Comment by zhenxiao** [comment]
@martint thank you for the comments, get it updated

File: presto-main/src/main/java/com/facebook/presto/sql/planner/EffectivePredicateExtractor.java [hunk 1]
```diff
@@ -278,9 +264,12 @@ private Expression deriveCommonPredicates(PlanNode node, Function<Integer, Colle
         for (int i = 0; i < node.getSources().size(); i++) {
             Expression underlyingPredicate = node.getSources().get(i).accept(this, null);
 
-            Iterable<Expression> equalities = FluentIterable.from(mapping.apply(i))
-                    .filter(not(symbolMatchesExpression()))
-                    .transform(entryToEquality());
+            List<Expression> equalities = mapping.apply(i).stream()
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/EffectivePredicateExtractor.java]: This is much harder to read. Leave the predicate methods (but convert them to JDK Predicate)

File: presto-main/src/main/java/com/facebook/presto/sql/planner/EffectivePredicateExtractor.java [hunk 2]
```diff
@@ -146,9 +129,12 @@ public Expression visitProject(ProjectNode node, Void context)
 
         Expression underlyingPredicate = node.getSource().accept(this, context);
 
-        Iterable<Expression> projectionEqualities = FluentIterable.from(node.getAssignments().entrySet())
-                .filter(not(symbolMatchesExpression()))
-                .transform(entryToEquality());
+        List<Expression> projectionEqualities = node.getAssignments().entrySet().stream()
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/EffectivePredicateExtractor.java]: Same comment as below

**Comment by zhenxiao** [comment]
@electrum thank you, get comments addressed

**Comment by zhenxiao** [comment]
@electrum @martint added a few more java8 stream api update

**Comment by zhenxiao** [comment]
rebased

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**